src/pages/Overview/OverviewPage.tsx

```diff
@@ -620,6 +620,37 @@ export class OverviewPage extends React.Component<OverviewProps, State> {
           namespaceActions.push(enableAction);
         }
       }
+      if (serverConfig.kialiFeatureFlags.istioUpgradeAction) {
+        const upgradeAction = {
+          isGroup: false,
+          isSeparator: false,
+          isDisabled: !canWrite,
+          title: 'Upgrade to Newer Istio',
+          action: (ns: string) => this.onUpgradeDowngradeIstio(ns, serverConfig.kialiFeatureFlags.istioUpgradeRevision)
+        };
+        const downgradeAction = {
+          isGroup: false,
+          isSeparator: false,
+          isDisabled: !canWrite,
+          title: 'Downgrade to Current Installed Istio',
+          action: (ns: string) => this.onUpgradeDowngradeIstio(ns, serverConfig.kialiFeatureFlags.istioCurrentRevision)
+        };
+        if (
+          nsInfo.labels &&
+          ((nsInfo.labels[serverConfig.istioLabels.injectionLabelRev] &&
+            nsInfo.labels[serverConfig.istioLabels.injectionLabelRev] === 'prod') ||
+            (nsInfo.labels[serverConfig.istioLabels.injectionLabelName] &&
+              nsInfo.labels[serverConfig.istioLabels.injectionLabelName] === 'enabled'))
+        ) {
+          namespaceActions.push(upgradeAction);
+        } else if (
+          nsInfo.labels &&
+          nsInfo.labels[serverConfig.istioLabels.injectionLabelRev] &&
+          nsInfo.labels[serverConfig.istioLabels.injectionLabelRev] === 'canary'
```

Comment:
Same here 'prod' and 'canary' are used in the example but it can be defined by the user [1]

[1] https://istio.io/latest/docs/reference/commands/istioctl/#istioctl-experimental-revision-tag-generate